src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatSelection.cs

```diff
@@ -22,7 +21,9 @@ public VSCommanding.CommandState GetCommandState(FormatSelectionCommandArgs args
 
         public bool ExecuteCommand(FormatSelectionCommandArgs args, CommandExecutionContext context)
         {
-            return TryExecuteCommand(args, context).Result;
+            var result = false;
+            _joinableTaskContext.Factory.Run(async () => result = await TryExecuteCommand(args, context).ConfigureAwait(true));
+            return result;
         }
 
         private async Task<bool> TryExecuteCommand(FormatSelectionCommandArgs args, CommandExecutionContext context)
```

Comment:
❗️ `TryExecuteCommandAsync` (needs the `Async` suffix)

❗️ The first line of this method needs to be the following:

```csharp
await _joinableTaskContext.Factory.SwitchToMainThreadAsync();
```

Reply:
reverted